### PR TITLE
Added Support for UWP

### PR DESCRIPTION
**Pull request by TheElixZammuto**
This PR upstreams the work done in the moonlight-common-c of moonlight-xbox.
Like the enet PR, it just disables some features due to the smaller API surface UWP has

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,7 +34,10 @@ endif()
 target_link_libraries(moonlight-common-c PRIVATE enet)
 
 if(MSVC)
-  target_compile_options(moonlight-common-c PRIVATE /W3 /wd4100 /wd4232 /wd5105 /WX)
+  target_compile_options(moonlight-common-c PRIVATE /W3 /wd4100 /wd4232 /wd5105)
+  if(NOT TARGET_UWP)
+	target_compile_options(moonlight-common-c PRIVATE /WX)
```
Comment (cgutman) [inline CMakeLists.txt:39]: What warnings are you seeing on UWP?

File: src/PlatformSockets.c [hunk 2]
```diff
@@ -14,18 +14,26 @@
 
 #if defined(LC_WINDOWS)
 
+# if defined(WINAPI_FAMILY) && WINAPI_FAMILY==WINAPI_FAMILY_APP
+# define LC_UWP
+# else 
+# define LC_WINDOWS_DESKTOP
+#endif
```
Comment (cgutman) [inline src/PlatformSockets.c:21]: Can you move this into Platform.h where `LC_WINDOWS` is also defined? We only use it here today, but I could see us using it other places eventually too.

File: CMakeLists.txt (hunk 1 above)
Comment (TheElixZammuto) [reply CMakeLists.txt]: It seems that it was a problem on some older builds and I kept it as-is. Reverting to upstream this file does not cause any problems, so I'm reverting this file

**Review by cgutman — APPROVED** [review]

**Comment by cgutman** [comment]
Merged by hand to resolve submodule conflicts

**Pull request closed without merging**